src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -388,14 +404,23 @@ public byte[] getEncoded() throws IOException {
             DerOutputStream tmp = new DerOutputStream();
 
             // encode encryption algorithm
-            algid.encode(tmp);
+            if (algid != null) {
+                algid.encode(tmp);
+            } else {
+                try {
+                    // Let's hope params has been initialized by now.
+                    AlgorithmId.get(params).encode(tmp);
```

Comment:
It looks like that you want to take care of the cases that the AlgorithmId did not initialize yet.  See comments in  AlgorithmId.